api/hypershift/v1beta1/azure.go

```diff
@@ -422,11 +422,11 @@ type AzurePlatformSpec struct {
 	// +immutable
 	SecurityGroupID string `json:"securityGroupID"`
 
-	// managedIdentities contains the managed identities needed for HCP control plane and data plane components that
-	// authenticate with Azure's API.
+	// azureAuthenticationConfig is the type of Azure authentication configuration to use to authenticate with Azure's
+	// Cloud API.
 	//
 	// +required
-	ManagedIdentities AzureResourceManagedIdentities `json:"managedIdentities"`
+	AzureAuthenticationConfig AzureAuthenticationConfiguration `json:"azureAuthenticationConfig"`
```

Comment:
Has this API shipped? You should create a tombstone that shows what `managedIdentities` looked like so that if we ever decided to reintroduce a field with this name, the serialization would be the same

Reply:
No one should be vendoring the code from a self-managed Azure perspective because we don't currently support it.

Only managed services in Red Hat are vendoring the API and we are coordinating the changes with them as there are no long lived clusters yet since the managed services part is still in development.

Reply:
> No one should be vendoring the code from a self-managed Azure perspective because we don't currently support it.

You cannot guarantee that no one will ever vendor a particular merged part of the code. Just because they shouldn't be doing it, doesn't stop them. Perhaps in the future you do support the self managed and someone decides to vendor, and for some reason, they pick up an older commit, boom, they're screwed